app/src/main/cpp/HandModels.cpp

```diff
@@ -0,0 +1,356 @@
+                 + localNorm3 * a_jointWeights.z
+                 + localNorm4 * a_jointWeights.w;
+  normal = normalize(u_model * localNorm);
+
+  v_color = calculate_light(normal, u_lights, u_material);
+}
+)SHADER";
+
+const char* sFragmentShader = R"SHADER(
+precision mediump float;
+
+varying vec4 v_color;
+
+void main() {
+  gl_FragColor = vec4(v_color.xyz, 1.0f);
+}
+)SHADER";
+
+}
+
+namespace crow {
+
+struct HandGLState {
+  uint32_t jointCount;
+  std::vector<vrb::Matrix> bindMatrices;
+
+  GLuint vertexCount;
+  GLuint vboPosition;
+  GLuint vboNormal;
+  GLuint vboJointIndices;
+  GLuint vboJointWeights;
+
+  GLuint indexCount;
+  GLuint iboIndices;
+};
+
+struct HandModels::State : public vrb::ResourceGL::State {
+  GLuint vertexShader;
+  GLuint fragmentShader;
+  GLuint program;
+  GLint aPosition;
+  GLint aNormal;
+  GLint aJointIndices;
+  GLint aJointWeights;
+  GLint uJointMatrices;
+  GLint uPerspective;
+  GLint uView;
+  GLint uModel;
+  std::vector<struct HandGLState> handGLState;
+  State()
+      : vertexShader(0)
+      , fragmentShader(0)
+      , program(0)
+      , aPosition(0)
+      , aNormal(0)
+      , aJointIndices(0)
+      , aJointWeights(0)
+      , handGLState({})
+  {}
+};
+
+HandModelsPtr
+HandModels::Create(vrb::CreationContextPtr& aContext) {
+  return std::make_shared<vrb::ConcreteClass<HandModels, HandModels::State> >(aContext);
+}
+
+void
+HandModels::StartFrame() {
+}
+
+void
+HandModels::Draw(const vrb::Camera& aCamera, const Controller& aController) {
+  if (!aController.hasHandMesh)
+    return;
+
+  if (!m.program) {
+    VRB_ERROR("HandModel::Draw FAILED!");
+    return;
+  }
+
+  const HandGLState& state = m.handGLState.at(aController.index);
+
+  VRB_GL_CHECK(glUseProgram(m.program));
+
+  const GLboolean enabled = glIsEnabled(GL_DEPTH_TEST);
+  if (!enabled)
+    VRB_GL_CHECK(glEnable(GL_DEPTH_TEST));
+
+  glBlendFunc(GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
+  glEnable(GL_BLEND);
+
+  VRB_GL_CHECK(glUniformMatrix4fv(m.uPerspective, 1, GL_FALSE, aCamera.GetPerspective().Data()));
+  VRB_GL_CHECK(glUniformMatrix4fv(m.uView, 1, GL_FALSE, aCamera.GetView().Data()));
+  // @FIXME: We don't apply any model transform for now, but eventually we could
+  // use `controller.transformMatrix`, if there is need.
+  vrb::Matrix modelMatrix = vrb::Matrix::Identity();
+  VRB_GL_CHECK(glUniformMatrix4fv(m.uModel, 1, GL_FALSE, modelMatrix.Data()));
+
+  assert(aController.meshJointTransforms.size() == state.jointCount);
+  std::vector<vrb::Matrix> jointMatrices;
+  jointMatrices.resize(state.jointCount);
+  for (int i = 0; i < state.jointCount; i++) {
+    jointMatrices[i] = aController.meshJointTransforms[i].PostMultiply(state.bindMatrices[i]);
+    VRB_GL_CHECK(glUniformMatrix4fv(m.uJointMatrices + i, 1, GL_FALSE, jointMatrices[i].Data()));
+  }
+
+  VRB_GL_CHECK(glBindBuffer(GL_ARRAY_BUFFER, state.vboPosition));
+  VRB_GL_CHECK(glEnableVertexAttribArray((GLuint)m.aPosition));
```

Comment:
Nit: as a general comment, do separate the type casting from the variable with a white space